src/plugins/autocompletion_providers/Vagrant.ts

```diff
@@ -0,0 +1,40 @@
+import {PluginManager} from "../../PluginManager";
+import {linedOutputOf} from "../../PTY";
+import {styles, Suggestion, contextIndependent} from "./Common";
+
+const commands = contextIndependent(async() => {
+    let lines: string[] = [];
+
+    try {
+        lines = await linedOutputOf("vagrant", ["list-commands"], process.env.HOME);
+    } catch (e) {
+        // skip if command is not exist
+    }
```

Comment:
Using exceptions as flow control is an antipattern. http://programmers.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why